olive/passes/pytorch/tensor_parallel_llama2.py

```diff
@@ -0,0 +1,230 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# Automatically distribute HF Llama model using Tensor Parallelism
+# --------------------------------------------------------------------------
+
+import logging
+import math
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from transformers.activations import ACT2FN
+from transformers.models.llama.modeling_llama import LlamaAttention, LlamaMLP, repeat_kv, rotate_half
```

Comment:
I think importing some specific modules from a dependency like this at top level makes the whole package incompatible with older versions of `transformers`. For example, `transformers.models.llama` doesn't exist before version 4.28.0. 

What if we make this file as a helper util and move the actual pass to `tensor_parallel.py` and make it import the functions from here only where needed? Or whatever structure sounds reasonable to you.

Reply:
Like https://github.com/microsoft/Olive/blob/main/olive/passes/pytorch/qat_utils.py and https://github.com/microsoft/Olive/blob/main/olive/passes/pytorch/trt_utils.py

Reply:
Valid concern. I'm ok to move it to a helper util file.
I expect in future we'll have PytorchTensorParallel pass variation for non-llama model also.

Reply:
Done!